### PR TITLE
Selenium SPS install tests

### PR DESCRIPTION
**Pull request by hargitayjpl**
## Purpose
This PR includes updates to comments for better understanding of pytest functions, adds new selenium tests for the Install of SPS, improves the output readability of selenium tests, and ensures selenium screenshots are ordered based on the test sequence.

## Proposed Changes
- [ADD] Selenium tests for the installation of SPS.
- [CHANGE] Updated comments in pytest functions for better clarity.
- [CHANGE] Modified selenium output to use 'checks' (✓) and 'X's for improved readability.
- [CHANGE] Implemented sorting of selenium screenshots in GitHub, corresponding to the order of tests.

## Issues
- The current implementation still reads the terminal output to verify the success of installations. Future updates will focus on checking an HTML element on the screen that indicates the status of the installation.

## Testing
- Test results are viewable available in our team's Slack channel for review.

### CODE REVIEW

File: nightly_tests/nightly_tests_ondemand/conftest.py [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
 
+# Function executed at the end of the test session. It writes the test results to a file.
 def pytest_sessionfinish(session, exitstatus):
     with open('makereport_output.txt', 'w') as f:
         print_table(session.results, file=f)
 
+# Function to print the test results in a table format. The output can be directed to a file.
 def print_table(results, file=None):
+    # ASCII symbols for pass and fail
+    pass_symbol = '✔'
```
Comment (galenatjpl) [inline nightly_tests/nightly_tests_ondemand/conftest.py:29]: Minor comment.  Usually I'm all for variable use, if it's used in multiple places, and can optimize runtime/memory.  However, as far as I see, these are only used on line 42 below.  I would rather just see these ascii symbols inline on line 42 for readability, and just get rid of lines 29 and 30.

File: nightly_tests/nightly_tests_ondemand/conftest.py [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
     name_width = max(max_name_length, len('Test Name'))
 
+    # Print table headers.
     print(f"\n{'Test Name'.ljust(name_width)} | {'Status'}", file=file)
     print(f"{'-' * name_width}-+--------", file=file)
 
+    # Print each test result in the table with the respective symbol.
     for result in results:
-        print(f"{result['name'].ljust(name_width)} | {result['status']}", file=file)
+        status_symbol = pass_symbol if result['status'] == 'PASSED' else fail_symbol
```
Comment (galenatjpl) [inline nightly_tests/nightly_tests_ondemand/conftest.py:42]: see comment above

**Review by galenatjpl — APPROVED** [review]
Looks great!  Just see the minor comment I made, that's all

File: nightly_tests/nightly_tests_ondemand/conftest.py (hunk 1 above)
Comment (hargitayjpl) [reply nightly_tests/nightly_tests_ondemand/conftest.py:29]: Agreed, makes sense. Incorporating it directing in its specific context will improve readability.

**Pull request merged**